### PR TITLE
Database setup

### PR DESCRIPTION
**Pull request by bhtibrewal**
test stats 0 0 0

### CODE REVIEW

File: skill-tree /pom.xml [hunk 1]
```diff
@@ -1,41 +1,35 @@
...
 		<dependency>
-			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-devtools</artifactId>
+			<groupId>com.mysql</groupId>
+			<artifactId>mysql-connector-j</artifactId>
 			<scope>runtime</scope>
-			<optional>true</optional>
-		</dependency>
-		<dependency>
-			<groupId>org.projectlombok</groupId>
```
Comment (vikhyat187) [inline skill-tree /pom.xml:36]: @bhtibrewal why is this removed?

Comment (vikhyat187) [reply skill-tree /pom.xml:36]: We will need this

File: skill-tree /pom.xml [hunk 2]
```diff
@@ -1,42 +1,40 @@
...
+	<name>skill-tree</name>
 	<description>skill tree project</description>
 	<properties>
 		<java.version>17</java.version>
 	</properties>
 	<dependencies>
 		<dependency>
 			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-hateoas</artifactId>
+			<artifactId>spring-boot-starter-data-jpa</artifactId>
```
Comment (iamitprakash) [inline skill-tree /pom.xml:22]: any specific reason to use spring-boot-starter-data-jpa

**Review by iamitprakash — APPROVED** [review]

File: skill-tree /pom.xml (hunk 2 above)
Comment (vikhyat187) [reply skill-tree /pom.xml:22]: Hi @bhtibrewal can you share the reason for this here, I wanted to understand what points to consider while choosing one.

**Review by vikhyat187 — COMMENTED** [review]
Just one comment, to share the thought process on how we decided to go with Spring data JPA and why not hibernate, rest looks good to me.

File: skill-tree /pom.xml (hunk 2 above)
Comment (bhtibrewal) [reply skill-tree /pom.xml:22]: as you suggested I read about it and understood that it's a data access layer, it removes the DAO implementation and boilerplate and provides the CRUD operations in a much simplified way.

references I read: https://www.baeldung.com/the-persistence-layer-with-spring-data-jpa
https://spring.io/projects/spring-data-jpa#overview

File: skill-tree /src/main/resources/application.properties [hunk 3]
```diff
@@ -1 +1,7 @@
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+spring.datasource.url=jdbc:mysql://localhost:3306/skilltree
+spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.MySQL8Dialect
+spring.datasource.username=${MYSQL_DB_USERNAME}
+spring.datasource.password=${MYSQL_DB_PASSWORD}
+spring.jpa.hibernate.ddl-auto=update
```
Comment (DashDeipayan) [inline skill-tree /src/main/resources/application.properties:6]: in prod, ddl rules must be `none`. 
It comes at a cost though.
Link: https://stackoverflow.com/questions/42135114/how-does-spring-jpa-hibernate-ddl-auto-property-exactly-work-in-spring

File: skill-tree /src/main/java/com/RDS/skilltree/model/Skill.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.RDS.skilltree.model;
+
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+
+@Entity
+public final class Skill { // TODO: to be updated according to the database
+    @Id
+    private Long id;
```
Comment (DashDeipayan) [inline skill-tree /src/main/java/com/RDS/skilltree/model/Skill.java:10]: Why is ID a long and not a UUID/ULID. There is no @GeneratedValue annotation as well

File: skill-tree /src/main/java/com/RDS/skilltree/SkillTreeApplication.java [hunk 5]
```diff
@@ -2,12 +2,20 @@
...
 
 @SpringBootApplication
+@RestController
 public class SkillTreeApplication {
 
 	public static void main(String[] args) {
 		SpringApplication.run(SkillTreeApplication.class, args);
 	}
 
+	@GetMapping("/")
```
Comment (DashDeipayan) [inline skill-tree /src/main/java/com/RDS/skilltree/SkillTreeApplication.java:17]: Routes must go to separate controller file

**Review by DashDeipayan — CHANGES_REQUESTED** [review]

File: skill-tree /src/main/resources/application.properties (hunk 3 above)
Comment (bhtibrewal) [reply skill-tree /src/main/resources/application.properties:6]: okay, we will be creating a separate application-prod.properties

File: skill-tree /src/main/java/com/RDS/skilltree/SkillTreeApplication.java (hunk 5 above)
Comment (bhtibrewal) [reply skill-tree /src/main/java/com/RDS/skilltree/SkillTreeApplication.java:17]: yes, that will be taken. up in the follow-up PRs this is just to have the confidence that the BE is up and running, and will be removed when we have proper APIs ready

File: skill-tree /src/main/java/com/RDS/skilltree/model/Skill.java (hunk 4 above)
Comment (bhtibrewal) [reply skill-tree /src/main/java/com/RDS/skilltree/model/Skill.java:10]: // TODO: to be updated according to the database
I have added the comment that we will not be using this model and update it according to the requirements

**Comment by bhtibrewal** [comment]
> Just one comment, to share the thought process on how we decided to go with Spring data JPA and why not hibernate, rest looks good to me.

had gone through this doc while reading on JPA, you go through it to understand the advantages and disadvantages listed., on based of which I decided one

**Review by DashDeipayan — APPROVED** [review]
Please create tickets for the issues that you proposed as a 'follow up'
